"dubious usage of method asInstanceOf" false positive

## Reproduction steps

Scala version: 2.13.16

```
$ cat Main.scala
object Main {
  def callFunction(function: PartialFunction[Unit, Unit]): Unit = function(???)
  def main(arguments: Array[String]): Unit = callFunction { _ => }
}

$ scalac -Wconf:any:error -Xlint:_ Main.scala
Main.scala:3: error: dubious usage of method asInstanceOf with unit value
Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=lint-universal-methods, site=Main.main
  def main(arguments: Array[String]): Unit = callFunction { _ => }
                                                              ^
1 error
```

## Problem

Arguably, `callFunction` should accept a `Unit => Unit`, but in reality, we discovered this bug with a generic method that accepts a `PartialFunction`, to which we wanted to provide an ordinary `Function`. I feel like this warning shouldn't be outputted in such a case.

--- a/src/compiler/scala/tools/nsc/typechecker/Typers.scala
+++ b/src/compiler/scala/tools/nsc/typechecker/Typers.scala
@@ -5484,7 +5484,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
             wrapErrors(t, _.typed1(t, mode, pt))
           }
           def checkDubiousUnitSelection(result: Tree): Unit =
-            if (!isPastTyper && isUniversalMember(result.symbol))
+            if (!isPastTyper && isUniversalMember(result.symbol) && result.pos.isRange)
               context.warning(tree.pos, s"dubious usage of ${result.symbol} with unit value", WarningCategory.LintUniversalMethods)
 
           val sym = tree.symbol

--- a/test/files/pos/t13134.scala
+++ b/test/files/pos/t13134.scala
@@ -1,0 +1,6 @@
+//> using options -Werror -Xlint
+
+object Main {
+  def callFunction(function: PartialFunction[Unit, Unit]): Unit = function(???)
+  def main(arguments: Array[String]): Unit = callFunction { _ => }
+}

